src/Analyzers/CSharp/Analyzers/UseExpressionBody/Helpers/UseExpressionBodyHelper`1.cs

```diff
@@ -118,36 +117,34 @@ protected virtual Location GetDiagnosticLocation(TDeclaration declaration)
         }
 
         protected virtual bool TryConvertToExpressionBody(
-            TDeclaration declaration,
-            ParseOptions options, ExpressionBodyPreference conversionPreference,
+            TDeclaration declaration, ExpressionBodyPreference conversionPreference,
             out ArrowExpressionClauseSyntax expressionWhenOnSingleLine,
             out SyntaxToken semicolonWhenOnSingleLine)
         {
             return TryConvertToExpressionBodyWorker(
-                declaration, options, conversionPreference,
+                declaration, conversionPreference,
                 out expressionWhenOnSingleLine, out semicolonWhenOnSingleLine);
         }
 
         private bool TryConvertToExpressionBodyWorker(
-            SyntaxNode declaration, ParseOptions options, ExpressionBodyPreference conversionPreference,
+            SyntaxNode declaration, ExpressionBodyPreference conversionPreference,
             out ArrowExpressionClauseSyntax expressionWhenOnSingleLine, out SyntaxToken semicolonWhenOnSingleLine)
         {
             var body = GetBody(declaration);
+            var languageVersion = ((CSharpParseOptions)body.SyntaxTree.Options).LanguageVersion;
```

Comment:
so i tehnically think this *may* not work.  but i'm not 100% certain.  i think the issue was that people can create nodes in isolation and then pass to helpers like this.  those nodes are *not* pointing at the right option.  indeed, they dont' point at syntaxtree at all (instead one gets synthesized by default).  This default syntax tree can have wrong 'default values', which can throw things off.

I'd be interested to see if any tests fail here.  If not, we may be ok.

Reply:
Revisited.